### DISCUSSION TITLE
Update Field Value with App State

### DISCUSSION THREAD

**Discussion by rhialto-de-schitterende** [Budibase/budibase#12397] [Help] [answered]
I have a form for creating titles (books).
On this form is a button New Author to create a new author if needed (normally author can be selected).
The Action Steps with this button are 1) Navigate To (form to create Author) and 2) Update Field Value to update the series_name field with the Prefix of the Author which is set in the App State authorPrefix in the create author form.
(There is no logic in writing the prefix in the series_name but I want to see if this works)
When I push the button and fill the fields in the new author form I there push te Create author button.
The Action Steps with this button are 1)  Validate form, 2) Save Row, 3 Update State (with form field Prefix) and 4) Close Screen Modal.
My title screen comes back but the field series_name is not filled with te prefix.
Then when I push te New Author butten again the series_name gets filled.
It looks like the Actions 1) Navigate To and 2. Update Field Value are processes at the same time instead of one by one.
Is this normal behavior or do I something wrong?

![image](https://github.com/Budibase/budibase/assets/151472393/930e05e1-70a3-4384-ac18-61629ff9af50)

![image](https://github.com/Budibase/budibase/assets/151472393/f90d52ed-0201-48e2-8a6d-9616cc73630c)

**Comment by ConorWebb96** [comment, marked as answer]
Hey @rhialto-de-schitterende,

If you are navigating between pages you will need to make sure [persist this value](https://docs.budibase.com/docs/app-state#persisting-values) is checked, in your bottom screenshot this is off meaning that any state set will be lost whenever the page changes.

I hope this makes sense.

**Reply by rhialto-de-schitterende** [reply]
Hi ConnerWebb96, 
Thanks for your reply. Persist this value doesn't solve my problem. Working with the App State does transfers the value to the previous form but the value is only shown the next time I push the button New Author. I presume that the Actions within the button New Author are all executed the same time when I push that button. So the actions Navigate To and Update Field Value are executed before transfering to the form in de Navigate To.
That is not what I expected and thereby not usable to transfer parameters between forms.
But maybe I'm wrong and is there still a way to do the job.

**Reply by ConorWebb96** [reply]
The actions attached to any component are run asynchronously, this means they will execute in the order they appear in. So if you have the `navigate to` action above the `update state` action this won't work as it will switch the page before it can update the state.

I can confirm that you can definitely transfer parameters/variables between forms on different pages. Between pages you would have to set the state on the default value of the field within the design view.
